src/jobs/membership.ts

```diff
@@ -0,0 +1,51 @@
+import cron from 'node-cron';
+import User from '../models/user';
+import { MEMBERSHIP_STATUS } from '../types/user';
+import {
+  GoogleSpreadsheet,
+  WorksheetBasicProperties,
+} from 'google-spreadsheet';
+import GoogleCreds from '../config/acs-web-5aaaf-768e669e2d39.json';
+
+const getTerm = () => {
+  const date = new Date();
+  const month = date.getMonth();
+  if (month === 1) {
+    return 'Fall' + (date.getFullYear() - 1);
+  } else if (month === 5) {
+    return 'Winter' + date.getFullYear();
+  } else if (month === 9) {
+    return 'Spring' + date.getFullYear();
+  }
+};
+
+const doc = new GoogleSpreadsheet(
+  '1Kn3Fc1vmi2dd9I645BxBQZMXXNvIJi1zEGnBiHuHDtk',
+);
+
+cron.schedule('0 0 1 JAN,MAY,SEP *', async function () {
+  try {
+    await doc.useServiceAccountAuth(GoogleCreds);
+    await doc.loadInfo();
+    const newSheet = await doc.addSheet({
+      title: getTerm(),
+    } as WorksheetBasicProperties);
+    const members = await User.find({
+      $or: [
+        { membershipStatus: MEMBERSHIP_STATUS.PAID },
+        { membershipStatus: MEMBERSHIP_STATUS.UNPAID },
+      ],
+    });
+    members.forEach(async (member) => {
+      const { name, email, watIAMUserId, membershipStatus } = member;
+      const newRow = { name, email, watIAMUserId, membershipStatus };
+      await newSheet.addRow(newRow);
+      Object.assign(member, {
+        membershipStatus: MEMBERSHIP_STATUS.EXPIRED,
+      });
```

Comment:
How come we don't just do `member.membershipStatus = MEMBERSHIP_STATUS.EXPIRED` here?

Reply:
I **honestly** can't remember why I did this🤣 I'll use `member.membershipStatus = MEMBERSHIP_STATUS.EXPIRED`